src/components/popup-components/AddToFavouritesButton/AddToFavouritesButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { useState } from "react"
+
+import { sendToBackground } from "@plasmohq/messaging"
+
+import "./AddToFavouritesButton.css"
+
+const AddToFavouritesButton = ({ popupId }) => {
+  const [status, setStatus] = useState<
+    "idle" | "loading" | "rejected"
+  >("idle")
+  const [isFavourite, setIsFavourite] = useState<boolean>(false)
+
+  const clickHandler = async () => {
+    setStatus("loading")
+    const { data, error } = await sendToBackground({
+      name: "updateUserDetails",
+      body: {
+        favourites: {
+          [isFavourite ? "disconnect" : "connect"]: [popupId]
+        }
+      }
+    })
+
+    if (error) {
+      console.error(error)
+      setStatus("rejected")
+    }
+
+    if (data) {
```

Comment:
Do we really need this evaluation?  Based on how we structured the returns on `fetchStrapiData` it wouldn't ever not return data if there were no error either